### DISCUSSION TITLE
Allow us to hide "Ship" or "Pick up" using the delivery customization hide operation

### DISCUSSION THREAD

**Discussion by ryanrphillips** [Shopify/function-examples#218] [Feedback: Delivery Customization API]
Hello!

We built a delivery and pickup date picker app and part of our app's flow is having the customer choosing a delivery method either on the product page or on the cart page.

At the moment, the customer is forced to select their delivery method again when they get to the checkout. We are looking for a way to pre-select this (or ideally force their selection) for them based on their selection in our widget to avoid them choosing one method in our app and a different method in the checkout (as this breaks the validity of the order).

The current delivery customization features allow us to hide shipping methods (the ones that show up once the customer chooses a delivery method), but not the delivery methods themselves.

Is this an operation that could possibly be added to the delivery customization functions?

Thank you!

Ryan

**Comment by rbarreca** [comment]
We also have a private app that lets users choosing delivery date and method before getting to checkout. We have very similar requirements as @ryanrphillips, except we'd prefer to also have the option just pre-select "Ship" vs "Pick up" and then pre-select the shipping or pickup location. See https://github.com/Shopify/function-examples/discussions/237.

We offer delivery every day and are using the new [Local Pickup Delivery Option Generator API](https://shopify.dev/docs/api/functions/reference/local-pickup-delivery-option-generator) to hide invalid pickup locations for that delivery date, so for now it's okay if they change method on the checkout page. Once we get our liquor license, it will be product dependent (e.g. no alcohol in orders going to certain pickup locations) so we'd possibly want the ability to force it and require user to go back to cart if they want to change.

**Comment by rockeynebhwani** [comment]
We have more use cases where this is helpful.

We are a US merchant shipping to international destinations with single Shopify instance. We don't have any stores in international markets. We want to hide 'Pick-up' option when user is in international context (based on currency OR market). 

We can do this in Checkout.liquid but not via Functions. 

This is an example from gorjana.com with exact same use case. When you access the site from UK, you see US locations with are 3000+ miles away. This is not ideal.

<img width="666" alt="Screenshot 2024-05-21 at 22 39 18" src="https://github.com/Shopify/function-examples/assets/1718757/0e17db3d-1292-454f-a69a-feaf556657c5">

**Comment by rockeynebhwani** [comment]
Another use case is 

- Clients wants to show 'Store availability' on PDP but operationally is not ready to support actual BOPIS in stores but happy for customers to see availability online and drive to store if they wants to

In order to this, we need to set locations to offer 'local pickup' (otherwise PDP store availability doesn't work) but we want to hide 'Pick-up' option in checkout till business is ready to offer BOPIS in stores operationally but it seems we can't do this today.

**Comment by dani-sanomads** [comment]
Do we have any updates for it? @rbarreca , @ryanrphillips

**Comment by bithaolee** [comment]
Any update?

**Comment by lerkens** [comment]
Here's our use case:

A store has a front end customization that allows customers to shop based on what is in stock at a given location. They have several retail locations and one warehouse location. The warehouse location is the only location with shipping options. Not all items are stocked at all locations.

In the scenario where a customer opts to shop at a retail location and they add items to cart that are not stocked at the "nationwide delivery" location, in checkout  they are presented with a preselected "Ship" option plus an error message, instead of a preselected "pickup" option (Even though there is an available pickup option, but no shipping option). It's a bottleneck because customers are seeing the error message and contacting support or abandoning cart, because they don't recognize that they can have to switch to "pickup" to see their pickup option.

We'd like to be able to default the cart to Pickup in this scenario.

**What customers see:**
![Screenshot 2024-08-27 at 17 55 54](https://github.com/user-attachments/assets/6bd62df5-c034-4931-9e85-130dc4d4324d)

**What we want them to see:**
![Screenshot 2024-08-27 at 17 56 31](https://github.com/user-attachments/assets/4675f66a-636b-4218-afd0-825cef39186e)

**Comment by blanklob** [comment]
Guys any update on this. Thanks

**Comment by colin-domaine** [comment]
Have another instance of a merchant looking to use location-specific inventory but who is not ready to implement BOPIS.

**Reply by rockeynebhwani** [reply]
It's now possible but you will have to contact MSM. We got this done for one merchant who was not ready to enable BOPIS in checkout but wanted to display store inventory on PDPs. Team had to enable a specific flag in back-end for us to do this.

**Reply by dani-sanomads** [reply]
How's that possible? @rockeynebhwani . and what's MSM?

**Reply by evaldas-raisutis** [reply]
Hey man, spill the juice 😂 what did they do exactly.

**Comment by pepsimax2k** [comment]
Also needing this - silly to have to select delivery/pickup in cart (to allow date/time slots to be chosen), then hope they both notice, and select, the same thing again in checkout. Simply not a viable solution.

**Comment by evaldas-raisutis** [comment]
Any news?

**Comment by jmanzo** [comment]
I ended up with a solution, but it's hacky and not scalable at all. It would be great to have a formal solution for this ASAP.

I use `delivery-customizations` to reorder or hide delivery options. I don't like that you'll need to disable the Pickup Option natively from Shopify. Also, based on what you need, you will be required to create a checkout using the storefront API. It's not an ideal solution at all, but it's workable so far.

**Reply by bansavage** [reply]
Were you able to hide the Ship option, or move Pick Up above Ship?

**Comment by Miseasd** [comment]
Any news about this. Really needed feature

**Comment by bansavage** [comment]
C'mon Shopify

**Comment by otherchris0782** [comment]
The shopify's pickup_point_delivery method type does not works for the current webhook version.
I found this kind of error occur while I was building the pickup auto webhook handler shopify custom app.
Does anyone had this kind of issue before?
If someone please let me know

**Comment by elitheasian** [comment]
Our store would also greatly appreciate this feature!
